### PR TITLE
ShareId submodule

### PR DESCRIPTION
**Pull request by SKOCHERI**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/userId/eids.js [hunk 1]
```diff
@@ -42,12 +42,12 @@ const USER_IDS_CONFIG = {
 
   // liveIntentId
   'lipb': {
-    getValue: function(data) {
+    getValue: function (data) {
```
Comment (coryhammon1) [inline modules/userId/eids.js]: Standard javascript style typically leaves out the space between the function keyword and the param list. We should be consistent with style that the rest of the project has.

File: test/spec/modules/userId_spec.js [hunk 2]
```diff
@@ -18,14 +18,15 @@ import {id5IdSubmodule} from 'modules/id5IdSystem.js';
...
+import {sharedIdSubmodule} from 'modules/sharedIdSystem.js';
 import {server} from 'test/mocks/xhr.js';
 
 let assert = require('chai').assert;
 let expect = require('chai').expect;
 const EXPIRED_COOKIE_DATE = 'Thu, 01 Jan 1970 00:00:01 GMT';
 
-describe('User ID', function() {
-  function getConfigMock(configArr1, configArr2, configArr3, configArr4, configArr5, configArr6) {
+describe('User ID', function () {
```
Comment (coryhammon1) [inline test/spec/modules/userId_spec.js]: Same as comment above.

**Review by coryhammon1 — COMMENTED** [review]
Just the small formatting things.

File: modules/sharedIdSystem.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+/**
+ * Checks if id needs to be synced
+ * @param configParams
+ * @param storedId
+ * @returns {boolean}
+ */
+function isIdSynced(configParams, storedId) {
+  var needSync = storedId.ns;
```
Comment (yava-ua) [inline modules/sharedIdSystem.js]: nit: do not use var, only const and in rare cases let. For code clarity
https://blog.usejournal.com/awesome-javascript-no-more-var-working-title-999428999994

File: modules/sharedIdSystem.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+const OPT_OUT_VALUE = '00000000000000000000000000';
+
+/**
+ * Constructs cookie value
+ * @param value
+ * @param needsSync
+ * @returns {string}
+ */
+function constructCookieValue(value, needsSync) {
+  let cookieValue = {};
```
Comment (yava-ua) [inline modules/sharedIdSystem.js]: why not opt for more strict approach:
```
const cookieValue = {
id: value,
ts: utils.timestamp(),
ns: needsSync === true
}
```

File: modules/sharedIdSystem.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+      let ns = (value && typeof value['ns'] === 'boolean') ? value['ns'] : undefined;
+      if (ns == undefined) {
+        bidIds.third = sharedId;
+      }
+      result.sharedid = bidIds;
+      utils.logInfo('SharedId: Decoded value ' + JSON.stringify(result));
+      return result;
+    }
+    return sharedId;
+  } catch (ex) {
```
Comment (yava-ua) [inline modules/sharedIdSystem.js]: what is it catching? seems can be a simple if check

File: modules/sharedIdSystem.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+  name: MODULE_NAME,
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {string} value
+   * @returns {{sharedid:{ 1: string, 3:string}} or undefined if value doesn't exists
+   */
+  decode(value) {
+    return ((value) ? encodeId(value) : undefined);
```
Comment (yava-ua) [inline modules/sharedIdSystem.js]: nit: no need for extra parentheses

File: modules/sharedIdSystem.js [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        if (responseBody) {
+          try {
+            let responseObj = JSON.parse(responseBody);
+            storedId = constructCookieValue(responseObj.sharedId, false);
+            utils.logInfo('SharedId: Older SharedId: ' + storedId.id);
+          } catch (error) {
+            utils.logError(error);
+          }
+        }
+      } catch (error) {
```
Comment (yava-ua) [inline modules/sharedIdSystem.js]: do you need this catch?

File: modules/sharedIdSystem.js (hunk 4 above)
Comment (SKOCHERI) [reply modules/sharedIdSystem.js]: "ns" is set only if /id endpoint fails and doesn't exists otherwise

**Review by coryhammon1 — APPROVED** [review]

**Pull request merged**